src/components/FeatureMenu/FeatureMenu.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import { compose } from 'lodash/fp';
+import { FaInfo } from 'react-icons/fa';
+import Map from 'ol/Map';
+import OLVectorLayer from 'ol/layer/Vector';
+import OLVectorSource from 'ol/source/Vector';
+import { setMenuOpen } from '../../model/app/actions';
+import FeaturePagination from '../FeaturePagination';
+import MenuItem from '../Menu/MenuItem';
+
+const propTypes = {
+  map: PropTypes.instanceOf(Map).isRequired,
+  title: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  icon: PropTypes.element,
+  renderBody: PropTypes.func.isRequired,
+
+  // mapStateToProps
+  clickedFeatureInfo: PropTypes.shape(),
+
+  // mapDispatchToProps
+  dispatchSetMenuOpen: PropTypes.func.isRequired,
+  t: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  className: null,
+  icon: <FaInfo />,
+  clickedFeatureInfo: null,
+};
+
+class FeatureMenu extends Component {
+  constructor(props) {
+    super(props);
+    const { map, clickedFeatureInfo } = this.props;
+
+    this.state = {
+      open: !!clickedFeatureInfo || false,
+      collapsed: true,
+      featureIndex: 0,
```

Comment:
You could control the index of the selected feature in the `FeaturePagination` component. As a consequence, it can be removed from the `FeatureMenu` and the `Popup` components. I think the whole `FeatureMenu` component is obsolete. The code will become very specific and can be moved to the `HandicapMenu`.